### ISSUE TITLE
Error initialising vaapi on Gnome Wayland

### ISSUE THREAD

**Issue by darren-steven** [biswasab/chromium_vaapi_fed#7]
I know this issue might be broader than just this app, but I think vaapi is working on other apps under wayland (vlc for example)

chromium-vaapi-70.0.3538.77-1.fc29.x86_64.rpm from rpmfusion koji
Hardware Intel Skylake

$ chromium-vaapi 
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[22955:22955:1116/104946.418564:ERROR:vaapi_wrapper.cc(328)] vaInitialize failed: unknown libva error

- Defaults to software decoder in youtube

Works fine in Gnome X11 session (I see GPU decoder selected in chrome://media-internals for youtube)

vaainfo reports same info on wayland and X11

$ vainfo 
libva info: VA-API version 1.3.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_3
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.3 (libva 2.3.0)
vainfo: Driver version: Intel i965 driver for Intel(R) Skylake - 2.2.0
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Simple            :	VAEntrypointEncSlice
      VAProfileMPEG2Main              :	VAEntrypointVLD

**Comment by qnixsynapse** [comment]
Hello.
Chromium runs through XWayland unlike vlc.
 See [link](https://github.com/intel/intel-vaapi-driver/pull/369)

I'm trying my best to compile chromium with wayland ozone but the build is still failing(complaining of a missing vaapi drm file). Unless any of this is solved, chromium on intel wayland won't able to decode in hardware.

Update:
Try Launching Chromium with --use-gl=egl

**Comment by maddie927** [comment]
That flag doesn't make a difference for me

**Comment by qnixsynapse** [comment]
@spicydonuts hmm. Anyway that PR is closed now. The only thing left to do is enabling ozone on chromium.

**Comment by rathann** [comment]
> @spicydonuts hmm. Anyway that PR is closed now. The only thing left to do is enabling ozone on chromium.

It is closed, but not merged due to submitter inactivity: https://github.com/intel/intel-vaapi-driver/pull/369#issuecomment-440376357 . It needs to be resubmitted.

**Comment by qnixsynapse** [comment]
@rathann Yes. Also there is a bug report @ [Bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1491186). It's been there since Fedora 26

**Comment by ghost** [comment]
> Hello.
> Chromium runs through XWayland unlike vlc.
> See [link](https://github.com/intel/intel-vaapi-driver/pull/369)
> 
> I'm trying my best to compile chromium with wayland ozone but the build is still failing(complaining of a missing vaapi drm file). Unless any of this is solved, chromium on intel wayland won't able to decode in hardware.
> 
> Update:
> Try Launching Chromium with --use-gl=egl

are there still problems with Ozone wayland?

**Comment by mthomos** [comment]
Any update on this issue ?

**Comment by qnixsynapse** [comment]
@mthomos I am providing chromium-ozone binaries for Fedora in [copr](https://copr.fedorainfracloud.org/coprs/akarshanbiswas/chromium-ozone/). These come with vaapi disabled. 

For intel vaapi driver there is still need for someone to add support for DRI3.

You can try building your own version of libva-intel-driver using these patches(for now):
[1](https://github.com/intel/intel-vaapi-driver/pull/369/commits/5db870533f6865903ff656b8cfa6b4da022773bb) and [2](https://github.com/intel/intel-vaapi-driver/pull/369/commits/a9e038879a62ad5614f8c318748448a23c3eec81)
